### PR TITLE
ESROGUE-539: Add accessor to set the SRP hardware timeout

### PR DESCRIPTION
**Pull request by mskoufis**

### CODE REVIEW

**Comment by ruck314** [comment]
@mskoufis How do I access `setHardwareTimeout()` from python?

**Comment by mskoufis** [comment]
@ruck314 I haven't implemented a python wrapper for this functionality.  Perhaps I should look at that.

**Comment by bengineerd** [comment]
The timeout value should be an instance variable of the SrpV3 class. The set function sets the instance variable and that gets used when building every header.

**Review by bengineerd — COMMENTED** [review]
The timeout value is stored in bits 31:24 of `header[0]`. You'll need to shift and mask `timeout` as you assign it into `header[0]`.

Also the style we use for instance variables is to end with `_`, so the variable should be named `timeout_`.

**Comment by mskoufis** [comment]
Okay, I had assumed that the shifted value would be provided to this function.  I'll work on that.

**Comment by bengineerd** [comment]
Ok, now let's expose `setHardwareTimeout()` to python. 
Master.cpp provides a pretty good example of how to do it.
https://github.com/slaclab/rogue/blob/main/src/rogue/interfaces/memory/Master.cpp#L46-L71

**Comment by bengineerd** [comment]
Also, I know it's a simple one-liner, but I would prefer that the `setHardwareTimeout()` implementation be in the .cpp file for style consistency.

**Comment by mskoufis** [comment]
Okay, no problem.

File: src/rogue/protocols/srp/SrpV3.cpp [hunk 1]
```diff
@@ -55,9 +55,9 @@ rps::SrpV3Ptr rps::SrpV3::create () {
 //! Setup class in python
 void rps::SrpV3::setup_python() {
 #ifndef NO_PYTHON
-
-   bp::class_<rps::SrpV3, rps::SrpV3Ptr, bp::bases<ris::Master,ris::Slave,rim::Slave>,boost::noncopyable >("SrpV3",bp::init<>());
-
+   bp::class_<rps::SrpV3, rps::SrpV3Ptr, bp::bases<ris::Master,ris::Slave,rim::Slave>,boost::noncopyable >("SrpV3",bp::init<>())
+      .def("_setHardwareTimeout",           &rps::SrpV3::setHardwareTimeout)
+   ;
```
Comment (ruck314) [inline src/rogue/protocols/srp/SrpV3.cpp:60]: Can you make the code and semi-colon on the "same line"

**Review by ruck314 — APPROVED** [review]

**Review by bengineerd — APPROVED** [review]

**Pull request merged**